rpmconf/Install/zmsetup.pl

```diff
@@ -2668,6 +2658,38 @@ sub setAdminPass {
   }
 }
 
+sub setLicenseCode {
+  if ($config{LICENSECODE} eq "") {
+    while (1) {
+      my $new = askNonBlank("Please enter the license code:",$config{LICENSECODE});
+      if (length($new) >= 18 && length($new) <= 24) {
+	      $config{LICENSECODE} = $new;
+	      return;
+      } else {
+	      print "Licensecode should be 18-24 digit!\n";
+      }
+    }
+  }
+}
+
+sub setLicenseActivationChoice {
```

Comment:
@RakeshAMore - for offline activation, we would need the daemon-service to be running in the offline mode right. is this the part where we ask the users that question and run the daemon service accordingly?

Reply:
Yes it is part of the daemon service installation

Reply:
Ok, along with this, we should also consider clearing the folder contents `/opt/zimbra/license/work/` folder while restarting the daemon service.